src/dmd/opover.d

```diff
@@ -1080,16 +1080,60 @@ Expression op_overload(Expression e, Scope* sc)
                 }
             }
 
+            bool partialInit;
+            Expression tempResult;
             if (!(e.op == TOK.assign && ad2 && ad1 == ad2)) // https://issues.dlang.org/show_bug.cgi?id=2943
             {
                 result = checkAliasThisForLhs(ad1, sc, e);
                 if (result)
-                    return;
+                {
+                    /* https://issues.dlang.org/show_bug.cgi?id=19441
+                     *
+                     * alias this may not be used for partial assignment.
+                     * If a struct has a single member which is aliased this
+                     * directly or aliased to a ref getter function that returns
+                     * the mentioned member, then alias this may be
+                     * used since the object will be fully initialised.
+                     * If the struct is nested, the context pointer is considered
+                     * one of the members, hence the `ad1.fields.dim == 2 && ad1.vthis`
+                     * condition.
+                     */
+                    if (e.op != TOK.assign)
```

Comment:
please do these to if's in one line.
```
if (e.op != TOK.assign || e.e1.op == TOK.type)
    return
```

the less return instructions are in the code, the better because that reduces explicit control-flow changes, `jmp` is easier to trace than `retn` is

Reply:
ok, done.